src/consensus/validation.h

```diff
@@ -22,6 +22,76 @@ static const unsigned char REJECT_NONSTANDARD = 0x40;
 static const unsigned char REJECT_INSUFFICIENTFEE = 0x42;
 static const unsigned char REJECT_CHECKPOINT = 0x43;
 
+/** A "reason" why something was invalid, suitable for determining whether the
+  * provider of the object should be banned/ignored/disconnected/etc.
+  * These are much more granular than the rejection codes, which may be more
+  * useful for some other use-cases.
+  */
+enum class ValidationInvalidReason {
+    // txn and blocks:
+    NONE,                    //!< not actually invalid
```

Comment:
I understand if you don't wanna bother this late in the process, but it really seems like `ValidationResult` or `ValidationOutcome` would have been easier on the eyes, where this would be `SUCCESSFUL` or something.

Reply:
I'd prefer to rename this in a future PR, since that would be a simple change and I don't want to hold up the structural improvements here.